File: Lista03_HerancaEPolimorfismo/PegadaCarbono/app/src/main/java/PegadaCarbono/Predio.java
package PegadaCarbono;

public abstract class Predio implements PegadaDeCarbono{
    private int nFrequentadores;
    private double mediaLixoPorFrequentador;
    private double consumoGasPorDia;

    public Predio(int nFrequentadores, double mediaLixoPorFrequentador, double consumoGasPorDia) {
        this.nFrequentadores = nFrequentadores;
        this.mediaLixoPorFrequentador = mediaLixoPorFrequentador;
        this.consumoGasPorDia = consumoGasPorDia;
    }

    public int getNFrequentadores() {
        return nFrequentadores;
    }

    public void setNFrequentadores(int nFrequentadores) {
        this.nFrequentadores = nFrequentadores;
    }

    public double getMediaLixoPorFrequentador() {
        return mediaLixoPorFrequentador;
    }

    public void setMediaLixoPorFrequentador(double mediaLixoPorFrequentador) {
        this.mediaLixoPorFrequentador = mediaLixoPorFrequentador;
    }

    public double getConsumoGasPorDia() {
        return consumoGasPorDia;
    }

    public void setConsumoGasPorDia(double consumoGasPorDia) {
        this.consumoGasPorDia = consumoGasPorDia;
    }

    public double getPegadaCarbono(){
        double pCarbono = (mediaLixoPorFrequentador * nFrequentadores * 0.2) + (consumoGasPorDia * 0.1);
        return pCarbono;
    }
}

File: Lista11_Python/Ex03_PegadaDeCarbono/predio.py
from abc import ABC, abstractmethod
from interfacePegadaDeCarbono import *

class Predio(PegadaDeCarbono, ABC):

    def __init__(self, nFrequentadores, mediaLixoPorFrequentador, consumoGasDiario):
        self.nFrequentadores = nFrequentadores
        self.mediaLixoPorFrequentador = mediaLixoPorFrequentador
        self.consumoGasDiario = consumoGasDiario

    @abstractmethod
    def __str__(self):
        strFreq = str(self.nFrequentadores)
        strMedLixo = str(self.mediaLixoPorFrequentador)
        strConsGas = str(self.consumoGasDiario)

        string = "nFrequentadores: "+strFreq+"\n"
        string = string+"mediaLixoPorFrequentador: "+strMedLixo+"\n"
        string = string+"consumoGasDiario: "+strConsGas

        return string

    def getNFrequentadores(self):
        return self.nFrequentadores

    def setNFrequentadores(self, nFrequentadores):
        self.nFrequentadores = nFrequentadores

    def getMediaLixoPorFrequentador(self):
        return self.mediaLixoPorFrequentador

    def getConsumoGasDiario(self):
        return self.consumoGasDiario

    @abstractmethod
    def getPegadaCarbono(self):
        pCarbono = (self.mediaLixoPorFrequentador * self.nFrequentadores * 0.2) + (self.consumoGasDiario * 0.1)
        return pCarbono
